src/components/landing-layout/footer/language-picker.tsx

```diff
@@ -40,17 +40,26 @@ const languages = {
   sk: 'Slovensky'
 }
 
-const findLanguageCode = (wantedLanguage: string): string => {
-  let foundLanguage = Object.keys(languages)
-                            .find((supportedLanguage) => wantedLanguage === supportedLanguage)
-  if (!foundLanguage) {
-    foundLanguage = Object.keys(languages)
-                          .find((supportedLanguage) => wantedLanguage.substr(0, 2) === supportedLanguage)
-  }
-  return foundLanguage || ''
-}
+/**
+ * This function checks if the wanted language code is supported by our translations.
+ * The language code that is provided by the browser can (but don't need to) contain the region.
+ * Some of our translations are region dependent (e.g. chinese-traditional and chinese-simplified).
+ * Therefore we first need to check if the complete wanted language code is supported by our translations.
+ * If not, then we look if we at least have a region independent translation.
+ *
+ * @param wantedLanguage an ISO 639-1 standard language code
+ */
+const findLanguageCode = (wantedLanguage: string): string => (
+  (
+    Object.keys(languages)
+          .find((supportedLanguage) => wantedLanguage === supportedLanguage)
+  ) ?? (
+    Object.keys(languages)
+          .find((supportedLanguage) => wantedLanguage.substr(0, 2) === supportedLanguage)
+  ) ?? ''
```

Comment:
Performance thing: Wouldn't it make sense to cache the return value of `Object.keys(languages)` to not calculate that twice?

Reply:
I wrapped it into a memo